frontend/src/views/ShareSecretPublicPage/ShareSecretPublicPage.tsx

```diff
@@ -46,18 +50,26 @@ export const ShareSecretPublicPage = () => {
 
   useEffect(() => {
     const updateTimer = () => {
-      if (data && data.expiresAt) {
-        const expirationTime = new Date(data.expiresAt).getTime();
-        const currentTime = new Date().getTime();
-        const timeDifference = expirationTime - currentTime;
+      if (data) {
+        if (data.expiresAt) {
+          const expirationTime = new Date(data.expiresAt).getTime();
+          const currentTime = new Date().getTime();
+          const timeDifference = expirationTime - currentTime;
 
-        if (timeDifference < 0) {
-          setTimeLeft("Expired");
-        } else {
-          const hoursRemaining = Math.floor((timeDifference % millisecondsPerDay) / millisecondsPerHour);
-          const minutesRemaining = Math.floor((timeDifference % millisecondsPerHour) / millisecondsPerMinute);
-          const secondsRemaining = Math.floor((timeDifference % millisecondsPerMinute) / 1000);
-          setTimeLeft(`${hoursRemaining}h ${minutesRemaining}m ${secondsRemaining}s`);
+          if (timeDifference < 0) {
+            setTimeLeft("Expired");
+          } else {
+            const hoursRemaining = Math.floor(
+              (timeDifference % millisecondsPerDay) / millisecondsPerHour
+            );
+            const minutesRemaining = Math.floor(
+              (timeDifference % millisecondsPerHour) / millisecondsPerMinute
+            );
+            const secondsRemaining = Math.floor((timeDifference % millisecondsPerMinute) / 1000);
+            setTimeLeft(`${hoursRemaining}h ${minutesRemaining}m ${secondsRemaining}s`);
+          }
+        } else if (data.expiresAfterViews) {
+          setTimeLeft(`${data.expiresAfterViews - 1} more views`);
```

Comment:
hmm.. I think we don't need to show this prompt to the public user? it's better that we limit the stuff they know about the shared secret

Reply:
aye nice idea! done for both 💪🏼